k2/csrc/properties.cc

```diff
@@ -55,6 +55,22 @@ bool IsTopSorted(const Fsa &fsa) {
   return true;
 }
 
+bool IsArcSorted(const Fsa &fsa) {
+  ArcLabelCompare comp;
+  StateId final_state = fsa.NumStates() - 1;
+  auto begin = fsa.arcs.begin();
+  const auto &arc_indexes = fsa.arc_indexes;
+  // we will not check the final state as it has no arcs leaving it.
+  for (StateId state = 0; state < final_state; ++state) {
+    // as non-empty `fsa` contains at least two states,
```

Comment:
Before saying this, you have to ensure that the passing fsa is not empty.

Reply:
It does not matter, as for empty fsa, `final_state` will be -1;